### DISCUSSION TITLE
How to configure micronaut to pickup/find @org.springframework.stereotype.Controller apis defined in 3rd party spring boot jar

### DISCUSSION THREAD

**Discussion by subash89** [micronaut-projects/micronaut-core#10871] [Q&A]
I have micronaut spring application following https://guides.micronaut.io/latest/spring-boot-to-micronaut-application-class-maven-java.html
1. When I invoke an api defined in a @org.springframework.stereotype.Controller  class in a 3rd party jar, it gives 404. How to configure micronaut to pick such spring controllers.
2. How openapi UI can pick up those 3rd party jar defined Controller apis ? May be it's related to #1

**Comment by yawkat** [comment]
In order to run a spring application using the micronaut framework, the source must be processed using the micronaut-spring annotation processors as shown [here](https://guides.micronaut.io/latest/micronaut-spring-boot-gradle-java.html). 3rd party jars (binary, not source) are not supported.

**Reply by graemerocher** [reply]
either that or you need to use `@Import`. See https://micronaut-projects.github.io/micronaut-spring/latest/guide/#sharingLibraries

**Reply by subash89** [reply]
@graemerocher  I added an @Import(com.my.other.Rest1Controller.class) in my Micronaut Application class. Where com.my.other.Rest1Controller.class is the 3rd party library controller class. But still 404. Does the import works for @Controller type too ?

**Reply by subash89** [reply]
Just FYI moving 3rd party controller and related bean classes into micronaut is not a viable/practical path for us.  Hence trying to find a way to get those compiled spring code run on micronaut. 

**In micronaut spring docs it says "You can take the compiled application and include it into another Spring or Micronaut application without change, which is a huge win for library authors."**  I thought my use case falls into this.

**Reply by graemerocher** [reply]
provide an example that reproduces the issue in an issue report

**Reply by subash89** [reply]
@graemerocher  I created https://github.com/micronaut-projects/micronaut-core/issues/10879 and uploaded a example two projects and provided steps.
(used same springboot and Micronaut-spring app example codes in micronaut docs with slight code changes)

**Reply by subash89** [reply]
@graemerocher  so in terms if using @Import, do you expect a Controller type can also be @import'ed ?
